troposphere/s3.py

```diff
@@ -126,6 +128,27 @@ def validate(self):
                     'properties on S3 Bucket Lifecycle Rule. Please use '
                     '"Transitions" since the former has been deprecated.')
 
+        if 'NoncurrentVersionTransition' in self.properties:
+            if 'NoncurrentVersionTransitions' not in self.properties:
+                # aws moved from a single transition to a list of them
+                # and deprecated 'NoncurrentVersionTransition', so let's
+                # just move it to the new property and not annoy the user.
+                self.properties['NoncurrentVersionTransitions'] = [
```

Comment:
@SonOfBytes See https://github.com/cloudtools/troposphere/blob/master/troposphere/dynamodb.py#L6 for an example of using warnings.  Thanks!